crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -398,7 +398,14 @@ def check_for_running_instances
           return
         end
 
-        cmd = "openstack server list --all-projects --long " \
+        insecure = ""
+        keystone = ::Node.find("run_list_map:keystone-server").first
+        if keystone[:keystone][:ssl][:insecure] ||
+            (controller[:nova][:ssl][:enabled] && controller[:nova][:ssl][:insecure])
+          insecure = "--insecure"
+        end
```

Comment:
I think ```if keystone[:keystone][:ssl][:insecure] || controller[:nova][:ssl][:insecure]``` should be enough at least that's what we do in the newer recipes.

But checking proposals will not make your life easier I think. They don't carry more information than what's available on the nodes.